distribution/tools/entitlement-runtime/src/main/java/org/elasticsearch/entitlement/runtime/checks/EntitlementChecksService.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.runtime.checks;
+
+import org.elasticsearch.entitlement.checks.EntitlementChecks;
+import org.elasticsearch.entitlement.checks.EntitlementProvider;
+import org.elasticsearch.entitlement.runtime.api.Entitlement;
+import org.elasticsearch.entitlement.runtime.api.FlagEntitlement;
+import org.elasticsearch.entitlement.runtime.api.NotEntitledException;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static java.util.Collections.newSetFromMap;
+import static org.elasticsearch.entitlement.runtime.internals.EntitlementInternals.isActive;
+import static org.elasticsearch.entitlement.runtime.internals.EntitlementInternals.isFrozen;
+
+/**
+ * Implementation of the {@link EntitlementChecks} interface.
+ * The trampoline module loads this object via SPI.
+ */
+public class EntitlementChecksService implements EntitlementChecks {
+    /**
+     * @return the same instance of {@link EntitlementChecksService} returned by {@link EntitlementProvider}.
+     */
+    public static EntitlementChecksService get() {
```

Comment:
This one returns the subtype because it offers more functionality than the `Checks` interface. The `Checks` contains only the `check` methods to be called from the instrumented code; the `Service` class also offers `grant`, `activate`, `freeze`, etc.